spanner/changestreams/src/main/java/com/example/spanner/changestreams/ChangeStreamSample.java

```diff
@@ -0,0 +1,240 @@
+      DatabaseAdminClient dbAdminClient, String instanceId, String databaseId, String tableName,
+      String changeStreamName) throws Exception {
+    System.out.println(
+        String.format("Updating database DDL to create table %s and change stream %s.",
+          tableName, changeStreamName));
+
+    OperationFuture<Void, UpdateDatabaseDdlMetadata> op = dbAdminClient
+        .updateDatabaseDdl(
+        instanceId,
+        databaseId,
+        Arrays.asList(
+          String.format("CREATE TABLE %s ("
+            + "  SingerId   INT64 NOT NULL,"
+            + "  FirstName  STRING(1024),"
+            + "  LastName   STRING(1024)"
+            + ") PRIMARY KEY (SingerId)", tableName),
+          String.format("CREATE CHANGE STREAM %s FOR %s", changeStreamName, tableName)),
+        null
+      );
+
+    op.get(TIMEOUT_MINUTES, TimeUnit.MINUTES);
+  }
+  // [END spanner_change_streams_sample_create_change_streams]
+
+  // [START spanner_change_streams_sample_query_change_streams]
+  public static void queryChangeStream(DatabaseClient dbClient, String tableName,
+                                       String changeStreamName) {
+    // Insert test data into the table.
+    System.out.println(
+        String.format("Inserting rows "
+          + "(1, singer_1_first_name, singer_1_last_name) and "
+          + "(2, singer_2_first_name, singer_2_last_name) "
+          + "into %s table.", tableName));
+
+    insertRows(dbClient, tableName);
+
+    final Timestamp startTimestamp = Timestamp.now();
+    // end = start + 30 seconds.
+    final Timestamp endTimestamp = Timestamp.ofTimeSecondsAndNanos(
+        startTimestamp.getSeconds() + 30, startTimestamp.getNanos());
+
+    final ChangeStreamRecordMapper changeStreamRecordMapper =
+        new ChangeStreamRecordMapper();
+
+    // Execute an initial query to get partition tokens.
+    System.out.println("Executing change stream initial query.");
+    // For initial query the partition token is null.
+    List<ChangeStreamRecord> initialQueryRecords = executeChangeStreamQueryAndPrint(
+        dbClient, changeStreamName, startTimestamp, endTimestamp, null,
+        changeStreamRecordMapper);
+
+    System.out.println("Executing change stream partition queries.");
+    for (ChangeStreamRecord record : initialQueryRecords) {
+      // Executes a partition query to print data records that we just inserted.
+      if (record instanceof ChildPartitionsRecord) {
+        ChildPartitionsRecord childPartitionsRecord = (ChildPartitionsRecord) record;
+        for (ChildPartition childPartition : childPartitionsRecord.getChildPartitions()) {
+          executeChangeStreamQueryAndPrint(
+              dbClient, changeStreamName, childPartitionsRecord.getStartTimestamp(), endTimestamp,
+              childPartition.getToken(), changeStreamRecordMapper);
+        }
+      } else {
+        throw new IllegalArgumentException("Got a record that's not ChildPartitionsRecord for "
```

Comment:
Can't we get Heartbeat records here? Maybe we should throw only if we get a DataChangeRecord?

Reply:
Done.